src/hotspot/share/gc/g1/g1FullCollector.hpp

```diff
@@ -90,6 +94,11 @@ class G1FullCollector : StackObj {
   G1CMBitMap*              mark_bitmap();
   ReferenceProcessor*      reference_processor();
 
+  void update_attribute_table(HeapRegion* hr);
+
+  bool is_in_pinned_or_closed(oop obj) const { return _region_attr_table.is_pinned_or_closed(cast_from_oop<HeapWord*>(obj)); }
+  bool is_in_closed(oop obj) const { return _region_attr_table.is_closed_archive(cast_from_oop<HeapWord*>(obj)); }
+
```

Comment:
Can't decide if it's worth adding a inline.hpp for those. Just throwing it out there in case you thought about it as well :)